[docs] Add expo-iap to In-App Purchases Documentation (#36048)
expo/expo: docs/pages/guides/in-app-purchases.mdx

/>

<BoxLink

>   title={<><CODE>react-native-iap</CODE> (Deprecated)</>}

Reviewer: ```suggestion
  title={<><CODE>react-native-iap</CODE> (deprecated)</>}
```

>   description="A legacy React Native library for in-app purchases. Superseded by expo-iap for Expo users."

Reviewer: ```suggestion
  description={<>A legacy React Native library for in-app purchases. Superseded by <CODE>expo-iap</CODE> for Expo apps.</>}
```

- Let's add inline syntax highlight for the library name. Also, since this library is not maintained, can you add a warning callout in the library's readme and its docs to indicate the same?
- "Expo apps" is more preferable than "users".

Also, I wonder if you'd like to update reactnative.directory for the same: https://reactnative.directory/?search=iap

>   title={<><CODE>react-native-iap</CODE> (deprecated)</>}
>   description={<>A legacy React Native library for in-app purchases. Superseded by <CODE>expo-iap</CODE> for Expo apps.</>}

Reviewer: ```suggestion
  title={
    <>
      <CODE>react-native-iap</CODE> (deprecated)
    </>
  }
  description={
    <>
      A legacy React Native library for in-app purchases. Superseded by <CODE>expo-iap</CODE> for
      Expo apps.
    </>
  }
  href="https://github.com/hyochan/react-native-iap"
  Icon={GithubIcon}
```

>   href="https://github.com/hyochan/react-native-iap"
>   Icon={GithubIcon}

Reviewer: ```suggestion
```

/>

<BoxLink
  title={<><CODE>expo-iap</CODE></>}

>   description="The official Expo module for in-app purchases, supporting managed workflows with dev-client."

Reviewer: ```suggestion
  description="A React Native library for in-app purchases that works with development builds."
```

- "managed workflows" is a deprecated term. We prefer using development builds to indicate that a library contains native API integration, and the developer needs to create a development build in order to use the library. I'd like to avoid it here as well.
- The "official" means that a library is maintained by the Expo team, which is not true in this case. Let's remove it.

<BoxLink
  title={
    <>
      <CODE>react-native-iap</CODE> (deprecated)
    </>
  }
  description={
    <>
      A legacy React Native library for in-app purchases. Superseded by <CODE>expo-iap</CODE> for
      Expo apps.
    </>
  }
  href="https://github.com/hyochan/react-native-iap"
  Icon={GithubIcon}
/>

> <BoxLink
>   title={<><CODE>expo-iap</CODE></>}
>   description="A React Native library for in-app purchases that works with development builds."
>   href="https://github.com/hyochan/expo-iap"

Reviewer: ```suggestion
<BoxLink
  title={
    <>
      <CODE>expo-iap</CODE>
    </>
  }
  description="A React Native library for in-app purchases that works with development builds."
  href="https://github.com/hyochan/expo-iap"```
